### ISSUE TITLE
Tier pricing tax shows only including tax

### ISSUE THREAD

**Issue by erjenrijnders** [magento/magento2#12225]
### Preconditions (*)
Magento 2.2, 2.3, 2.4-develop

### Steps to reproduce (*)

1. Go to Sales > Tax >Price display settings.
2. Configure to show the including and excluding tax price.
3. Go to a configurable product on the frontend with tier pricing.
4. You will see tier price including tax.
5. There is no way to change that to excluding price.

### Expected result (*)
An extra option which one has first priority, including or excluding tax.

### Actual result (*)
![screenshot_1](https://user-images.githubusercontent.com/22836339/32751647-edc03ddc-c8c6-11e7-8f1c-9ac0f791ec00.png)
The tier price only shows the including tax while I need exactly the opposit. Excluding price first, also for tier pricing.

**Comment by erjenrijnders** [comment]
What does that mean "G1 Failed"? Sorry I see that label a lot but I have no idea what that means

**Comment by ishakhsuvarov** [comment]
@erjenrijnders Label G1 Failed means that not all of the required sections in the issue report are filled. You may find more information in this document: https://github.com/magento/magento2/wiki/Magento-Issue-Gates

**Comment by magento-engcom-team** [comment]
@erjenrijnders, thank you for your report.
We've created internal ticket(s) MAGETWO-83996 to track progress on the issue.

**Comment by erjenrijnders** [comment]
Hello,

This one is open more then a year now and I am still waiting for this change/feature to be honest. Now 2.3 is out but this task is still open.
Can you tell me maybe when you think this one will be done?

Thanks a lot

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29654` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by zeshanrahman** [comment]
<?php 
                            if($_product->getTierPrice()){
                                $tierPrice = $_product->getTierPrice();
                                     foreach ($tierPrice as $key => $value) {
                                    $qty = (int)$value['price_qty'];
                                   
                                    $price = $value['price'];

                                     $formattedTierPrice = $this->helper('Magento\Framework\Pricing\Helper\Data')->currency(number_format($price, 2), true, false);
                                    $savePercentage = round(100 - ( (100 / $_product->getPrice())* $price) ) ."%";
                                    echo "<div style='font-weight: 600;background: #f5f5f5;padding: 5px; display: block; margin: 20px 0px 0px'>"."Buy $qty for "."<span style='color: #000;font-size: 16px;'>" . $formattedTierPrice . "</span>"." each and save ".$savePercentage."<br>"."</div>";
                                }
                            }
                            ?>

**Comment by jasonujmaalvis** [comment]
Hi all, after coming across this issue the following fix appears to be working, you'll have to create a module to override the core function, don't edit the core.

### Backend

Location:

`module-configurable-product/Block/Product/View/Type/Configurable.php`

New line to show Ex VAT:

`'basePrice' => $this->localeFormat->getNumber($tierPrice['price']->getBaseAmount()),`

Core function with change added in:

```
    /**
     * Collect price options
     *
     * @return array
     */
    protected function getOptionPrices()
    {
        $prices = [];
        foreach ($this->getAllowProducts() as $product) {
            $tierPrices = [];
            $priceInfo = $product->getPriceInfo();
            $tierPriceModel =  $priceInfo->getPrice('tier_price');
            $tierPricesList = $tierPriceModel->getTierPriceList();
            foreach ($tierPricesList as $tierPrice) {
                $tierPrices[] = [
                    'qty' => $this->localeFormat->getNumber($tierPrice['price_qty']),
                    'price' => $this->localeFormat->getNumber($tierPrice['price']->getValue()),
                    'basePrice' => $this->localeFormat->getNumber($tierPrice['price']->getBaseAmount()),
                    'percentage' => $this->localeFormat->getNumber(
                        $tierPriceModel->getSavePercent($tierPrice['price'])
                    ),
                ];
            }
      ...
      ...
```
### Frontend

Location:

`Magento_ConfigurableProduct/templates/product/price/tier_price.phtml`

Template:

```
<script type="text/x-magento-template" id="tier-prices-template">
    <ul class="prices-tier items">
        <% _.each(tierPrices, function(item, key) { %>
        <%  var priceStr = '<span class="price-container price-tier_price">'
                + '<span data-price-amount="' + priceUtils.formatPrice(item.basePrice, currencyFormat) + '"' + ' data-price-type=""' + ' class="price-wrapper ">'
                    + '<span class="price">' + priceUtils.formatPrice(item.basePrice, currencyFormat) + '</span>'
                + '</span>'
                + '<span data-price-amount="' + priceUtils.formatPrice(item.price, currencyFormat) + '"' + ' data-price-type=""' + ' class="price-wrapper ">'
                    + '<span class="price">' + priceUtils.formatPrice(item.price, currencyFormat) + '</span>'
                + '</span>'
            + '</span>'; %>
        <li class="item">
            <%= '<?= $block->escapeHtml(__('Buy %1 for %2 each and', '%1', '%2')) ?>'
            .replace('%1', item.qty)
            .replace('%2', priceStr) %>
            <strong class="benefit">
                <?= $block->escapeHtml(__('save')) ?><span
                        class="percent tier-<%= key %>">&nbsp;<%= item.percentage %></span>%
            </strong>
        </li>
        <% }); %>
    </ul>
</script>
<div data-role="tier-price-block"></div>
```

**Comment by VladimirZaets** [comment]
Hi @erjenrijnders . Thanks for collaboration. This issue was reviewed on Community Contribution Triage meeting. The recording of the meeting you can look by the [link](https://www.youtube.com/watch?v=ce2JOgB2NbQ&list=PLDvMskiz68Q2BLkj0-xzjuFl7ZVti9Quj&index=3&t=0s)

**Comment by magento-engcom-team** [comment]
Hi @erjenrijnders. Thank you for your report. 
The issue has been fixed in magento/magento2#27953 by @jiten-patel in 2.4-develop branch
Related commit(s):
 - [c9bb1d39a4f367e95e38a3e8d54e0aa2d8120ea8](https://github.com/magento/magento2/commit/c9bb1d39a4f367e95e38a3e8d54e0aa2d8120ea8)
 - [0b839e9d296bc5f76ccec215ea8562bafce65a74](https://github.com/magento/magento2/commit/0b839e9d296bc5f76ccec215ea8562bafce65a74)
 - [53d72269f3d108135cd7df9ba262f8aa276013c0](https://github.com/magento/magento2/commit/53d72269f3d108135cd7df9ba262f8aa276013c0)
 - [e4be19bc290ecc45251c30a8e5a4d86676e531ba](https://github.com/magento/magento2/commit/e4be19bc290ecc45251c30a8e5a4d86676e531ba)
 - [406b1f4bf33c2502c8e40589c1cd092b173c8f23](https://github.com/magento/magento2/commit/406b1f4bf33c2502c8e40589c1cd092b173c8f23)
 - [f18aaf616998ec9b7b2235c543ae7447d9c4b0a4](https://github.com/magento/magento2/commit/f18aaf616998ec9b7b2235c543ae7447d9c4b0a4)
 - [1a78b49a9d7585dc9a3e5bd501d632ccaa873046](https://github.com/magento/magento2/commit/1a78b49a9d7585dc9a3e5bd501d632ccaa873046)
 - [182edf6e52d4e2f574d80b9e7ea6738acfffe2b1](https://github.com/magento/magento2/commit/182edf6e52d4e2f574d80b9e7ea6738acfffe2b1)
 - [bc6c2d68dee18beba4ededed3d2a91fbdec7ca62](https://github.com/magento/magento2/commit/bc6c2d68dee18beba4ededed3d2a91fbdec7ca62)
 - [59d5d3093c97485691f96d888b18d92132c39e64](https://github.com/magento/magento2/commit/59d5d3093c97485691f96d888b18d92132c39e64)
 - [659a6d9c06e253816674464026ef74dd26c6e4f3](https://github.com/magento/magento2/commit/659a6d9c06e253816674464026ef74dd26c6e4f3)
 - [4bed3cbf91b11b035e9494a493e282b3ff9fffbe](https://github.com/magento/magento2/commit/4bed3cbf91b11b035e9494a493e282b3ff9fffbe)
 - [c440566dd58217ef7d3d1cd3b72be7b72236b0b6](https://github.com/magento/magento2/commit/c440566dd58217ef7d3d1cd3b72be7b72236b0b6)

The fix will be available with the upcoming 2.4.2 release.
